storage/client/src/main/java/com/azure/storage/file/DirectoryAsyncClient.java

```diff
@@ -3,77 +3,527 @@
+            .map(this::getPropertiesResponse);
     }
 
-    public Mono<Response<DirectoryProperties>> getProperties(String shareSnapshot) {
-        throw new UnsupportedOperationException();
+    /**
+     * Sets the user-defined metadata to associate to the directory.
+     *
+     * <p>If {@code null} is passed for the metadata it will clear the metadata associated to the directory.</p>
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * <p>Set the metadata to "directory:updatedMetadata"</p>
+     *
+     * <pre>
+     * client.setMetadata(Collections.singletonMap("directory", "updatedMetadata"))
+     *     .subscribe(response -&gt; System.out.printf("Setting the directory metadata completed with status code %d", response.statusCode()));
+     * </pre>
+     *
+     * <p>Clear the metadata of the directory</p>
+     *
+     * <pre>
+     * client.setMetadata(null)
+     *     .subscribe(response -&gt; System.out.printf("Clearing the directory metadata completed with status code %d", response.statusCode()));
+     * </pre>
+     *
+     * @param metadata Optional. Metadata to set on the directory, if null is passed the metadata for the directory is cleared
+     * @return information about the directory
+     * @throws StorageErrorException If the directory doesn't exist or the metadata contains invalid keys
+     */
+    public Mono<Response<DirectorySetMetadataInfo>> setMetadata(Map<String, String> metadata) {
+        return azureFileStorageClient.directorys().setMetadataWithRestResponseAsync(shareName, directoryName, null, metadata, Context.NONE)
+            .map(this::setMetadataResponse);
     }
 
-    public Mono<Response<DirectoryInfo>> setMetadata(Map<String, String> metadata) {
-        throw new UnsupportedOperationException();
+    /**
+     * Lists all directories and files in the storage account without their prefix or maxResult.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * <p>List all directories and files in the account</p>
+     *
+     * <pre>
+     * client.listFilesAndDirectories()
+     *     .subscribe(result -&gt; System.out.printf("The file or directory %s exists in the account", result.name()));
+     * </pre>
+     *
+     * @return {@link FileRef File info} in the storage directory
+     */
+    public Flux<FileRef> listFilesAndDirectories() {
+        return listFilesAndDirectories(null, null);
     }
 
-    public Flux<FileRef> listFilesAndDirectories(String prefix, int maxResults, String shareSnapshot) {
-        throw new UnsupportedOperationException();
+    /**
+     * Lists all shares in the storage account with their prefix or snapshots.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * <p>List all directories with "subdir" prefix and return 10 results in the account</p>
+     *
+     * @codesnippet com.azure.storage.file.directoryAsyncClient.listFilesAndDirectories
```

Comment:
Generate the JavaDoc and review that the code snippet is visible in all places. I suspect this won't show up as expected.